script.js

```diff
@@ -1,4 +1,75 @@
+var numOfPlayers = 3;
+var player = [3, 2, 1];
+var playerInputMessage = `please input "1" or "2" for the order of the dice`;
+var playerNumbers = [];
+var outputValue = "";
+var gameMode = "starting";
+var rng = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+var diceRoll1 = rng();
+var diceRoll2 = rng();
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "starting") {
+    switch (numOfPlayers) {
+      case 3:
+        numOfPlayers = numOfPlayers - 1;
+        outputValue = `player 1 dice roll one is ${diceRoll1} and dice roll two is ${diceRoll2}, ${playerInputMessage} `;
+        gameMode = "input 1 or 2";
+        return outputValue;
+      case 2:
+        diceRoll1 = rng();
+        diceRoll2 = rng();
+        numOfPlayers = numOfPlayers - 1;
+        outputValue = `player 2 dice roll one is ${diceRoll1} and dice roll two is ${diceRoll2}, ${playerInputMessage} `;
+        gameMode = "input 1 or 2";
+        return outputValue;
+      case 1:
+        diceRoll1 = rng();
+        diceRoll2 = rng();
+        numOfPlayers = numOfPlayers - 1;
+        outputValue = `player 3 dice roll one is ${diceRoll1} and dice roll two is ${diceRoll2}, ${playerInputMessage} `;
+        gameMode = "input 1 or 2";
+        return outputValue;
+      case 0:
+        var maxValue = Math.max(...playerNumbers);
+        var maxIndex = playerNumbers.indexOf(
+          String(Math.max(...playerNumbers))
+        );
+        console.log(maxIndex);
+        var playingPlayer = [1, 2, 3];
+        console.log(playingPlayer);
+        console.log(playerNumbers);
+        var winningPlayer = playingPlayer[maxIndex];
+        outputValue = `The winning number is ${maxValue} by player ${winningPlayer}`;
+        return outputValue;
+    }
+  }
+  if (gameMode == "input 1 or 2") {
+    if (input == 1) {
+      var numberArray = player;
+      var players = Math.min(...numberArray);
```

Comment:
I don't really think it's completely necessary to use a `Math.min()` in this instance, the player that is the lowest is the last index of the array, so you can just do something like `player[player.length-1]`, or if you want to be fancy and use a built in array function, `player.at(-1)`.

The at function functions similarly to the square bracket notation, but the key difference is that, if you put in a negative number there, it counts backwards from the end of the array, so:
`player[player.length-2]` is the same as `player.at(-2)`